### PR TITLE
community selection modal: layout changes

### PR DESCRIPTION
**Pull request by jennur**
Closes https://github.com/inveniosoftware/invenio-app-rdm/issues/1463

## Community modal changes
- Changed text when no community selected
- Triggered modal closing on selecting community
- Added remove button to community header
- Removed modal footer
- Removed "Currently selected" section in modal
- Changed styling of selected community item
- Added link to "View community" instead of having the title be the link
- When clicking "Selected" in the modal, the community remains selected as it can be removed from the header
- Added accessibility

## Screenshots
Screenshot 2022-04-26 at 11 15 54
Screenshot 2022-04-26 at 11 26 22
Screenshot 2022-04-26 at 11 15 50
Screenshot 2022-04-26 at 11 16 02

### CODE REVIEW

File: src/lib/components/CommunitySelectionModal/CommunityListItem.js [hunk 1]
```diff
@@ -12,70 +12,57 @@ import PropTypes from 'prop-types';
...
 
   return (
-    <Item key={result.id} className="pr-20">
+    <Item key={result.id} className={itemSelected ? 'selected' : ''}>
       <Item.Image
-        as={Image}
         size="tiny"
-        src={linkToLogo}
-        fallbackSrc="/static/images/square-placeholder.png"
+        src={linkToLogo || "/static/images/square-placeholder.png"}
```
Comment (zzacharo) [inline src/lib/components/CommunitySelectionModal/CommunityListItem.js:28]: Why was this needed? I think the fallbackSrc is actually there in case the linkToLog is giving an error...Doesn't work as expected?

Comment (jennur) [reply src/lib/components/CommunitySelectionModal/CommunityListItem.js:28]: No, the fallback image didn't show up? But this solved the issue.

File: src/lib/components/Communities/CommunityHeader.js [hunk 2]
```diff
@@ -45,30 +45,31 @@ class CommunityHeaderComponent extends Component {
               </>
             ) : (
               <div className="page-subheader-element">
-                {i18next.t('No community selected.')}
+                {i18next.t('Select the community where you want to upload your record.')}
```
Comment (kpsherva) [inline src/lib/components/Communities/CommunityHeader.js:48]:
```suggestion
                {i18next.t('Select the community where you want to submit your record.')}
```
unless it was discussed and agreed in the mockups before, then please ignore this suggestion

File: src/lib/components/CommunitySelectionModal/CommunityListItem.js (hunk 1 above)
Comment (kpsherva) [reply src/lib/components/CommunitySelectionModal/CommunityListItem.js:28]: I think the fallback image is broken in the original component and this is only the workaround. It should be fixed in the source, the Image component of invenio-react-forms was implemented to automatically solve the issue of default image placeholder

File: src/lib/components/CommunitySelectionModal/CommunityListItem.js [hunk 3]
```diff
@@ -12,70 +12,57 @@ import PropTypes from 'prop-types';
...
-        <Item.Description>
+        <Item.Description as="p">
           {_truncate(metadata.description, { length: 150 })}
         </Item.Description>
         <Item.Extra>{metadata.type}</Item.Extra>
       </Item.Content>
-      <div className="flex">
-        {!standAlone && (
+      <Item.Extra className="flex width auto">
+        <div className="align-self-center">
```
Comment (kpsherva) [inline src/lib/components/CommunitySelectionModal/CommunityListItem.js:52]: why was the conditional statement `!standAlone` removed ?

File: src/lib/components/CommunitySelectionModal/CommunityListItem.js [hunk 4]
```diff
@@ -12,70 +12,57 @@ import PropTypes from 'prop-types';
...
+            rel="noreferrer"
+            size="small"
+            className="transparent pt-0 ml-15 mb-5"
+            content={i18next.t("View community")}
+            icon="external alternate"
+            title={i18next.t("Opens in new tab")}
+          />
         </Item.Header>
-        <Item.Description>
+        <Item.Description as="p">
```
Comment (kpsherva) [inline src/lib/components/CommunitySelectionModal/CommunityListItem.js:46]: this will most likely remove the SUI styling for Item.Description, to be verified

File: src/lib/components/CommunitySelectionModal/CommunityListItem.js (hunk 1 above)
Comment (jennur) [reply src/lib/components/CommunitySelectionModal/CommunityListItem.js:28]: Will change it back!

File: src/lib/components/CommunitySelectionModal/CommunityListItem.js (hunk 4 above)
Comment (jennur) [reply src/lib/components/CommunitySelectionModal/CommunityListItem.js:46]: The SUI styling is still applied :)

File: src/lib/components/CommunitySelectionModal/CommunitySelectionFooter.js [hunk 5]
```diff
@@ -24,4 +24,4 @@ export const CommunitySelectionFooter = () => {
       </Container>
     </>
   );
-};
+};
```
Comment (Pineirin) [inline src/lib/components/CommunitySelectionModal/CommunitySelectionFooter.js:27]: Why this change?

File: src/lib/components/CommunitySelectionModal/CommunityListItem.js (hunk 3 above)
Comment (Pineirin) [reply src/lib/components/CommunitySelectionModal/CommunityListItem.js:54]: :+1:

**Review by Pineirin — APPROVED** [review]

File: src/lib/components/CommunitySelectionModal/CommunitySelectionFooter.js (hunk 5 above)
Comment (jennur) [reply src/lib/components/CommunitySelectionModal/CommunitySelectionFooter.js:27]: I first committed changes to the file, but changed it back, and this happened. There is no change, so I don't know why it's showing up like this?

File: src/lib/components/CommunitySelectionModal/CommunityListItem.js (hunk 1 above)
Comment (zzacharo) [reply src/lib/components/CommunitySelectionModal/CommunityListItem.js]: @jennur Is it working after that change or the image is broken?

File: src/lib/components/CommunitySelectionModal/CommunitySelectionFooter.js (hunk 5 above)
Comment (kpsherva) [reply src/lib/components/CommunitySelectionModal/CommunitySelectionFooter.js:27]: its missing new line on the end of the file (please configure/run prettier)

**Review by kpsherva — APPROVED** [review]

File: src/lib/components/CommunitySelectionModal/CommunityListItem.js (hunk 1 above)
Comment (zzacharo) [reply src/lib/components/CommunitySelectionModal/CommunityListItem.js]: I checked the PR locally and the images are not shown properly. The issue should not be fixed here so I created an issue https://github.com/inveniosoftware/react-invenio-deposit/issues/482 . @jennur let me know if you want to investigate on this :) I will merge this PR as it is blocking https://github.com/inveniosoftware/invenio-app-rdm/issues/1483 .
